src/olympia/devhub/tasks.py

```diff
@@ -414,16 +415,46 @@ def run_addons_linter(path, listed=True):
     if not listed:
         args.append('--self-hosted')
 
-    process = subprocess.Popen(
-        args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    if not os.path.exists(path):
+        raise ValueError(
+            'Path "{}" is not a file or directory or does not exist.'
+            .format(path))
+
+    file_size = os.path.getsize(path)
+
+    if file_size > settings.FILE_UPLOAD_MAX_MEMORY_SIZE:
```

Comment:
I think you are going to need your own setting with a more sensible value designed for the validator. `FILE_UPLOAD_MAX_MEMORY_SIZE` is probably too big.